controllers/rhmi/bootstrapReconciler.go

```diff
@@ -496,3 +511,67 @@ func generateSecret(length int) string {
 	}
 	return string(buf)
 }
+
+func (r *Reconciler) processQuota(installation *rhmiv1alpha1.RHMI, namespace string,
+	installationQuota *quota.Quota, serverClient k8sclient.Client) error {
+	isQuotaUpdated := false
+
+	quotaParam, found, err := addon.GetStringParameterByInstallType(context.TODO(), serverClient, rhmiv1alpha1.InstallationTypeManagedApi, namespace, addon.QuotaParamName)
+	if err != nil {
+		return fmt.Errorf("error checking for quota secret %w", err)
+	}
+
+	quotaParam, err = getSecretQuotaParam(installation, quotaParam, found)
+	if err != nil {
+		return err
+	}
+
+	// get the quota config map from the cluster
+	configMap := &corev1.ConfigMap{}
+	err = serverClient.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: quota.ConfigMapName}, configMap)
+	if err != nil {
+		return fmt.Errorf("error getting quota config map %w", err)
+	}
+
+	// if both are toQuota and Quota are empty this indicates that it's either
+	// the first reconcile of an installation or it's the first reconcile of an upgrade to 1.6.0
+	// if the secretname is not the same as status.Quota this indicates there has been a quota change
+	// to an installation which is already using the Quota functionality.
+	// if either case is true set toQuota in the rhmi cr and update the status object and set isQuotaUpdated to true
+	if (installation.Status.ToQuota == "" && installation.Status.Quota == "") ||
+		quotaParam != installation.Status.Quota {
+		installation.Status.ToQuota = quotaParam
+		isQuotaUpdated = true
+	}
+
+	// Updates the installation quota to the quota param if the quota is updated
+	err = quota.GetQuota(quotaParam, configMap, installationQuota, isQuotaUpdated)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getSecretQuotaParam(installation *rhmiv1alpha1.RHMI, quotaParam string, found bool) (string, error) {
+
+	if found || quotaParam != "" {
```

Comment:
Should this if statement be checking that the `quotaParam` is found _and_ that it doesn't contain an empty value? Right now if `found` is True then the function will return without checking if `quotaParam` actually contains a value.
```suggestion
	if found && quotaParam != "" {
```

Reply:
Okay just had to look into this more and I think you are right, if this is left in it might even revert the quota values to "" when they are valid so I think this does need to be changed.